### PR TITLE
Additional object explorer scopes

### PR DESCRIPTION
**Pull request by NSExceptional**
Scopes now include:
- No inheritance (base class)
- Base class with parent attributes
- Base class with all inherited attributes, except NSObject
- NSObject attributes only

It is unusual that you need to see anything but the parent's attributes alongside those of the base class, and you especially rarely need to see NSObject attributes.

[Here's what it looks like.](https://i.imgur.com/sXMt7xT.png) The `+ Inherited` and `+ Parent` tabs hide depending on how far down the object is in the inheritance hierarchy (so, if it inherits directly from `NSObject`, the only other tab will be `NSObject`)

### CODE REVIEW

File: Classes/ObjectExplorers/FLEXObjectExplorerViewController.m [hunk 1]
```diff
@@ -294,11 +376,14 @@ - (void)updateIvars
...
     self.ivars = [[self class] ivarsForClass:class];
-    self.inheritedIvars = [[self class] inheritedIvarsForClass:class];
+    self.ivarsWithParent = [self.ivars arrayByAddingObjectsFromArray:[[self class] ivarsForClass:[class superclass]]];
+    self.inheritedIvars = [self.ivars arrayByAddingObjectsFromArray:[[self class] inheritedIvarsForClass:class]];
+    self.NSObjectIvars = [[self class] ivarsForClass:[NSObject class]];
 }
 
 + (NSArray<FLEXIvarBox *> *)ivarsForClass:(Class)class
 {
+    if (!class) return @[];
```
Comment (ryanolsonk) [inline Classes/ObjectExplorers/FLEXObjectExplorerViewController.m]: nit: curly braces always

File: Classes/ObjectExplorers/FLEXObjectExplorerViewController.m [hunk 2]
```diff
@@ -373,29 +456,36 @@ - (void)updateMethods
...
 - (void)updateFilteredClassMethods
 {
-    self.filteredClassMethods = [self filteredMethodsFromMethods:self.classMethods inheritedMethods:self.inheritedClassMethods areClassMethods:YES];
+    NSArray<FLEXMethodBox *> *candidateMethods = [self metadata:FLEXMetadataKindClassMethods forScope:self.scope];
+    self.filteredClassMethods = [self filteredMethodsFromMethods:candidateMethods areClassMethods:YES];
 }
 
 + (NSArray<FLEXMethodBox *> *)methodsForClass:(Class)class
 {
+    if (!class) return @[];
```
Comment (ryanolsonk) [inline Classes/ObjectExplorers/FLEXObjectExplorerViewController.m]: nit: curly braces always

File: Classes/ObjectExplorers/FLEXObjectExplorerViewController.m [hunk 3]
```diff
@@ -36,25 +50,30 @@ @interface FLEXMethodBox : NSObject
 @implementation FLEXMethodBox
 @end
 
-static const NSInteger kFLEXObjectExplorerScopeNoInheritanceIndex = 0;
-static const NSInteger kFLEXObjectExplorerScopeIncludeInheritanceIndex = 1;
-
 @interface FLEXObjectExplorerViewController () <UISearchBarDelegate>
```
Comment (ryanolsonk) [inline Classes/ObjectExplorers/FLEXObjectExplorerViewController.m]: I think we could get the structure of the state on this class to a better place but it's not urgent. I'd also like to kill the inheritance used to customize the object explorers and do it via configuration. Would be cool to expose public hooks on the framework so you could customize display for different classes.

**Comment by NSExceptional** [comment]
Ah yes, I forgot to fix the braces. I'm used to same-line braces.

Regarding object explorers, I think I commented about this on another issue but maybe not: I think objects should conform to some protocol that provides a custom explorer view controller if desired. That seems like a sleek solution to me

**Pull request merged**